python/lsst/analysis/tools/actions/vector/selectors.py

```diff
@@ -272,3 +277,71 @@ def getInputSchema(self) -> KeyedDataSchema:
 
     def __call__(self, data: KeyedData, **kwargs) -> Vector:
         return cast(Vector, data[self.vectorKey.format(**kwargs)])
+
+
+class AndSelector(VectorAction):
+
+    selectors = ConfigurableActionStructField[VectorAction](
+        doc="Selectors for selecting rows, will be AND together",
+    )
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        for action in self.selectors:
+            yield from action.getInputSchema()
+
+    def __call__(self, data: KeyedData, **kwargs) -> KeyedData:
+        mask: Optional[np.ndarray] = None
```

Comment:
if possible use `Vector` instead of `np.ndarray`, as it will be more portable whenever I get to making a proper Protocol for it

Reply:
Also, I have not been consistent in other places, I think I only did it in `SacalarActions` but I had a thought that possible someone might pass `mask=<some_mask>` in kwargs, possibly some extra mask that has been defined somewhere, like a notebook or something. Do you think it is worth it to change this to `mask: Vector | None = kwargs.get('mask')` ?

Reply:
This got fixed along the way.